webviews/welcome-page/src/components/WelcomePage.tsx

```diff
@@ -0,0 +1,35 @@
+import { cn, Flex } from '@axonivy/ui-components';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import logo from '../axonivy-logo.svg';
+import { useVscode } from '../util/useVscode';
+import { GettingStartedCommands } from './GettingStartedCommands';
+import { ShortcutSection } from './shortcut/ShortcutSection';
+import { SocialMediaLinks } from './SocialMediaLinks';
+import './WelcomePage.css';
+
+export const WelcomePage = () => {
+  const { t } = useTranslation();
+  const [initData, setInitData] = useState({ version: '' });
+  const { getData } = useVscode();
+  getData(setInitData);
```

Comment:
I whould just do this (pseudo code):

```suggestion
  useEffect(() => {
    const event = window.addEventListener('message', event => {
      setInitData(event);
    });
    return () => window.removeEventListener(event);
  }, []);
```